src/samVisualize/rendering/navigation.ts

```diff
@@ -0,0 +1,45 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as vscode from 'vscode'
+import * as yaml from 'yaml'
+/**
+ * Maps a resource name to the line number in the template in which it is defined
+ */
+export type ResourceLineMap = {
+    [index: string]: number
```

Comment:
string:number is a very generic type, no need for a special nominal type for this. If one doesnt already exist, we could add a generic `NameNumberPair` type in a file `shared/types.ts` .

Reply:
After refactoring to start/end positions the type is more specific. I think it's good to have now.